File: data/imported/knowledge/slides/net/merged/snippets/snippet_021.cs
[Fact]
    public void TestCoreProperties()
    {
        // Core properties persist after save/reload.
        using var pres = new Presentation();
        var props = pres.DocumentProperties;
        props.Title = "My Presentation";
        props.Subject = "Demo Subject";
        props.Author = "John Doe";
        props.Keywords = "demo, test";
        props.Category = "Examples";

        using var pres2 = TestHelpers.SaveAndReopen(pres, _tempDir);
        var p2 = pres2.DocumentProperties;
        p2.Title.Should().Be("My Presentation");
        p2.Subject.Should().Be("Demo Subject");
        p2.Author.Should().Be("John Doe");
        p2.Keywords.Should().Be("demo, test");
        p2.Category.Should().Be("Examples");
    }

File: data/imported/knowledge/slides/java/merged/snippets/snippet_021.java
@Test
    void testCoreProperties() throws IOException {
        try (var pres = new Presentation()) {
            IDocumentProperties props = pres.getDocumentProperties();
            props.setTitle("My Presentation");
            props.setSubject("Demo Subject");
            props.setAuthor("John Doe");
            props.setKeywords("demo, test");
            props.setCategory("Examples");

            try (var pres2 = saveAndReopen(pres)) {
                IDocumentProperties p2 = pres2.getDocumentProperties();
                assertThat(p2.getTitle()).isEqualTo("My Presentation");
                assertThat(p2.getSubject()).isEqualTo("Demo Subject");
                assertThat(p2.getAuthor()).isEqualTo("John Doe");
                assertThat(p2.getKeywords()).isEqualTo("demo, test");
                assertThat(p2.getCategory()).isEqualTo("Examples");
            }
        }
    }